test/evergreen.yml

```diff
@@ -154,6 +154,25 @@ tasks:
           configure_env_vars: CC=/opt/mongodbtoolchain/v3/bin/gcc CXX=/opt/mongodbtoolchain/v3/bin/g++ CFLAGS="-fsanitize=undefined -ggdb"
           posix_configure_flags: --enable-silent-rules --enable-strict --enable-diagnostic
 
+  # build with --disable-shared, without --enable-diagnostic
+  # the compiler builds turn on diagnostics
+  - name: compile-disable-shared-without-enable-diagnostic
+    tags: ["pull_request"]
+    commands:
+      - func: "get project"
+      - func: "compile wiredtiger"
+        vars:
+          posix_configure_flags: --enable-silent-rules --enable-strict --disable-shared
+
+  # build with --disable-static, without --enable-diagnostic
+  # the compiler builds turn on diagnostics
+  - name: compile-disable-static-without-enable-diagnostic
```

Comment:
I feel the `-without-enable-diagnostic` part could be saved for the task name, not strongly opinioned though.

Reply:
updated